src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -0,0 +1,100 @@
+package com.rybka.service;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.rybka.dao.HibernateDAO;
+import com.rybka.model.Currency;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class ExchangeService {
+
+    private static final String API_KEY = "4e0bbdc44fcdf05612fa0882";
+
+    private Double currencyValue;
+    private String baseCurrencyAbbreviation;
+    private String targetCurrencyAbbreviation;
+    private Double amount;
+    private Double total;
+
+    public void loadCurrencyOf(String userBaseCurrencyAbbreviation, String userTargetCurrencyAbbreviation) {
+        String url_str = String.format("https://prime.exchangerate-api.com/v5/%s/latest/%s",
+                API_KEY, userBaseCurrencyAbbreviation);
+
+        try {
+            URL url = new URL(url_str);
+            HttpURLConnection request = (HttpURLConnection) url.openConnection();
+            request.connect();
+
+            JsonParser jp = new JsonParser();
```

Comment:
Move logic to the separate methods. 

sendRequest()
retrieveJsonData()
constructDataObject()

Reply:
fixed